### PR TITLE
Refactoring: Changing print and exit to raise error

### LINKED ISSUE #330: Use of print() & exit() instead of exceptions
I just noticed that the code contains `exit()` and `print()` in what should be library code.

That should be replaced.
For example, this pattern is found a lot in https://github.com/ivadomed/ivadomed/blob/master/ivadomed/models.py#L1152

```patch
-                print('Model file not found: {}'.format(fname_model))
-                exit()
+                raise FileNotFoundError(fname_model)
```

### PR DESCRIPTION
**Pull request by lrouhier**
Changing old print and exit for "raise *Error*". This was found in a few files and is now fixed. 
fix #330

### CODE REVIEW

File: ivadomed/loader/adaptative.py [hunk 1]
```diff
@@ -703,8 +703,8 @@ def HDF5_to_Bids(HDF5, subjects, path_dir):
     hdf5 = h5py.File(HDF5, "r")
     # check the dir exists:
     if not path.exists(path_dir):
-        print("Directory doesn't exist. Stopping process.")
-        exit(0)
+        raise FileNotFoundError("Directory doesn't exist. Stopping process.")
```
Comment (charleygros) [inline ivadomed/loader/adaptative.py:706]: I would recommend printing the `path_dir` to facilitate debugging :-)

File: ivadomed/training.py [hunk 2]
```diff
@@ -405,10 +405,10 @@ def get_scheduler(params, optimizer, num_epochs=0):
         scheduler = optim.lr_scheduler.CyclicLR(optimizer, **params, mode="triangular2", cycle_momentum=False)
         step_scheduler_batch = True
     else:
-        print(
+        raise ValueError(
             "Unknown LR Scheduler name, please choose between 'CosineAnnealingLR', 'CosineAnnealingWarmRestarts',"
             "or 'CyclicLR'")
```
Comment (charleygros) [inline ivadomed/training.py:410]:
```suggestion
            "Unknown LR Scheduler name, please choose between 'CosineAnnealingLR', 'CosineAnnealingWarmRestarts',"
            " or 'CyclicLR'")
```

Comment (charleygros) [reply ivadomed/training.py:410]: I would recommend printing the unknown LR sch. name

File: ivadomed/training.py [hunk 3]
```diff
@@ -430,8 +430,7 @@ def get_loss_function(params):
                                "BinaryCrossEntropyLoss", "TverskyLoss", "FocalTverskyLoss", "AdapWingLoss", "L2loss",
                                "LossCombination"]
     if loss_name not in loss_function_available:
-        print("Unknown Loss function, please choose between {}".format(loss_function_available))
-        exit()
+        raise ValueError("Unknown Loss function, please choose between {}".format(loss_function_available))
```
Comment (charleygros) [inline ivadomed/training.py:433]: I would recommend printing the unknown loss function name

File: ivadomed/transforms.py [hunk 4]
```diff
@@ -115,9 +115,8 @@ def __init__(self, dict_transforms, requires_undo=False):
                 params_cur = {k: parameters[k] for k in parameters if k != "applied_to" and k != "preprocessing"}
                 transform_obj = globals()[transform](**params_cur)
             else:
-                print('ERROR: {} transform is not available in your ivadomed package. '
+                raise ValueError('ERROR: {} transform is not available in your ivadomed package. '
```
Comment (charleygros) [inline ivadomed/transforms.py:118]:
```suggestion
                raise ValueError('ERROR: {} transform is not available. '
```

File: ivadomed/training.py (hunk 3 above)
Comment (lrouhier) [reply ivadomed/training.py]: done in 
a3e884d

File: ivadomed/training.py (hunk 2 above)
Comment (lrouhier) [reply ivadomed/training.py]: done in 
a3e884d

File: ivadomed/loader/adaptative.py (hunk 1 above)
Comment (lrouhier) [reply ivadomed/loader/adaptative.py]: done in 
a3e884d

**Review by charleygros — APPROVED** [review]
Great, many thanks
